packages/core/src/server/service.test.ts

```diff
@@ -57,19 +57,29 @@ const graphqlSchema = buildGqlSchema(schema);
 const setup = async ({
   common,
   userStore,
+  options = {
+    hasCompletedHistoricalSync: true,
+  },
 }: {
   common: Common;
   userStore: UserStore;
+  options?: {
+    hasCompletedHistoricalSync?: boolean;
+  };
 }) => {
   await userStore.reload({ schema });
 
   const service = new ServerService({ common, userStore });
   await service.start();
   service.reload({ graphqlSchema });
 
+  if (options.hasCompletedHistoricalSync) {
+    service.setIsHistoricalEventProcessingComplete();
```

Comment:
Quick note on naming. At the moment, some of the terminology in the codebase is not internally consistent. I recently came to the conclusion that there are two components: "sync" and "indexing", which are relatively clear in the [architecture diagram](https://link.excalidraw.com/l/9zcSSq4IYVb/2nIu3McwAQP).

Specific to this PR, it'd be nice to rename "HistoricalEventProcessing" -> "HistoricalIndexing" (a find and replace across the repo should work). And, you should rename the option here to match (hasCompletedHistoricalIndexing).

Soon we'll do a wholesale renaming PR. This will also change the public terminology: "event handlers" -> "indexing functions".